contracts/tokenomics/maker/src/contract.rs

```diff
@@ -367,24 +450,35 @@ fn swap(
         return Ok(SwapTarget::Astro(msg));
     }
 
-    // 2. check if bridge token exists
-    let bridge_token = BRIDGES
-        .load(deps.storage, from_token.to_string())
-        .map_err(|_| ContractError::CannotSwap(from_token.clone()))?;
+    let uusd = native_asset_info(UUSD_DENOM.to_string());
+    let uluna = native_asset_info(ULUNA_DENOM.to_string());
+
+    if from_token.eq(&uluna) {
+        let uusd_pool = uusd_pool_exists(deps, cfg, from_token.clone());
+        if uusd_pool.is_none() {
+            return Err(ContractError::InvalidBridgeNoPool(
+                uluna.clone(),
+                uusd.clone(),
+            ));
+        }
 
+        let msg = build_swap_msg(deps, cfg, uusd_pool.unwrap(), from_token, amount_in)?;
+
+        return Ok(SwapTarget::Bridge { asset: uusd, msg });
+    }
+
+    let bridge_token = BRIDGES.load(deps.storage, from_token.to_string())?;
     let bridge_pool = query_pair_info(
         &deps.querier,
         cfg.factory_contract.clone(),
         &[from_token.clone(), bridge_token.clone()],
-    )
-    .map_err(|_| ContractError::InvalidBridgeNoPool(from_token.clone(), bridge_token.clone()))?;
-
-    let msg = build_swap_msg(deps, cfg, bridge_pool, from_token, amount_in)?;
+    );
```

Comment:
It is better to use `?` here and remove unwrap in 477 since panic is worse than descriptive Error message.